presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/NodeSelectionUtils.java

```diff
@@ -28,7 +29,8 @@ private NodeSelectionUtils() {}
 
     static List<HostAddress> sortedNodes(NodeMap nodeMap)
     {
-        return nodeMap.getNodesByHostAndPort().values().stream()
+        Collection<InternalNode> nodes = nodeMap.isResilientSoftAffinityEnabled() ? nodeMap.getCachedNodesByHostAndPort().values() : nodeMap.getNodesByHostAndPort().values();
```

Comment:
Since `cachedNodesByHostAndPort` is constructed by `cache#asMap`, do we need to worry that this is not updated frequent enough that the cache entries might expire?

Reply:
We are creating `cachedNodesByHostAndPort` everytime we create `NodeMap` . So i think it's as up to date as possible.